src/base/bittorrent/torrenthandle.h

```diff
@@ -445,14 +445,26 @@ namespace BitTorrent
 
         QHash<QString, TrackerInfo> m_trackerInfos;
 
-        enum StartupState
+        enum ProcessingState
         {
-            NotStarted,
-            Starting,
-            Started
+            PreparingPaused, // torrent is preparing to start processing in "paused" state
+            PreparingResumed, // torrent is preparing to start processing in "resumed" state
+            Starting, // torrent is prepared and starting to perform regular processing
+            Working, // torrent is performing regular processing
+            CheckingPaused // torrent is checking in "paused" state
         };
+        // Handle "extra" states of torrent processing.
+        // Initially torrent can be either in "PreparingPaused" or "PreparingResumed" state.
+        // Torrent is become "paused" after checking to perform all needed preparations.
+        // Torrent that is prepared to start processing in "paused" state becomes "Working"
+        // immediatelly after the preparations are done.
+        // Torrent that is prepared to start processing in "resumed" state becomes "Starting"
```

Comment:
Why the name `Starting` & `Working` are chosen? what about using `WorkingPaused` & `WorkingResumed`? I think this is clearer that the torrent is in working condition and is paused/resumed.

Also I think `resumed` postfix term is not so straightforward (sounds like it is coming from a previous paused state), perhaps `running` is better.

Reply:
>what about using WorkingPaused & WorkingResumed? I think this is clearer that the torrent is in working condition and is paused/resumed.

You misinterpret its meaning. "Starting" doesn't mean "Working paused".
Don't confuse high-level torrent states provided by qBittorrent with low-level (natuve) torrent states, provided by libtorrent.
>Also I think resumed postfix term is not so straightforward (sounds like it is coming from a previous paused state), perhaps running is better.

This term is used across the qBittorrent. I dislike it too, but when I suggested other options, they were all rejected.